src/config.rs

```diff
@@ -33,39 +33,59 @@ const ZTUNNEL_WORKER_THREADS: &str = "ZTUNNEL_WORKER_THREADS";
 
 const DEFAULT_WORKER_THREADS: usize = 2;
 
-#[derive(Clone, Debug)]
+#[derive(serde::Serialize,Clone, Debug)]
 pub struct Config {
+    #[serde(default)]
     pub window_size: u32,
+    #[serde(default)]
     pub connection_window_size: u32,
+    #[serde(default)]
     pub frame_size: u32,
 
+    #[serde(default)]
     pub socks5_addr: SocketAddr,
+    #[serde(default)]
     pub admin_addr: SocketAddr,
+    #[serde(default)]
     pub inbound_addr: SocketAddr,
+    #[serde(default)]
     pub inbound_plaintext_addr: SocketAddr,
+    #[serde(default)]
     pub outbound_addr: SocketAddr,
 
     /// The name of the node this ztunnel is running as.
+    #[serde(default)]
     pub local_node: Option<String>,
 
     /// CA address to use. If fake_ca is set, this will be None.
     /// Note: we do not implicitly use None when set to "" since using the fake_ca is not secure.
+    #[serde(default)]
     pub ca_address: Option<String>,
     /// XDS address to use. If unset, XDS will not be used.
+    #[serde(default)]
     pub xds_address: Option<String>,
     /// Filepath to a local xds file for workloads, as YAML.
+    #[serde(default)]
```

Comment:
why do we need this? isn't it for deserialization, nor serialization?

Reply:
hmm, maybe?  I thought I hit an error but looks like it's not needed.  I know `pub auth: identity::AuthSource` required the `#[serde(skip_serializing)]` annotation since it lacks a Serialize implementation